edb/schema/indexes.py

```diff
@@ -91,14 +91,39 @@ def is_index_valid_for_type(
         case 'fts::index':
             return is_subclass_or_tuple(expr_type, 'fts::document', schema)
         case 'pg::gist':
-            return expr_type.is_range() or expr_type.is_multirange()
+            return (
+                expr_type.is_range()
+                or
+                expr_type.is_multirange()
+                or
+                expr_type.issubclass(
+                    schema,
+                    (
+                        schema.get('ext::postgis::geometry',
+                                   type=s_scalars.ScalarType,
+                                   default=None),
+                        schema.get('ext::postgis::geography',
+                                   type=s_scalars.ScalarType,
+                                   default=None),
```

Comment:
Maybe schema types should be able to declare a list of indexes that are known to support them?